main_test.go

```diff
@@ -115,32 +115,69 @@ func TestParseFlagsError(t *testing.T) {
 type processGeofeedTest struct {
 	gf string
 	db string
-	dl string
+	dl []string
 	c  counts
+	em string
 }
 
 func TestProcessGeofeed(t *testing.T) {
-	tests := []processGeofeedTest{
+	goodTests := []processGeofeedTest{
 		{
 			"test_data/geofeed1.csv",
 			"test_data/GeoIP2-City-Test.mmdb",
-			"Found a potential improvement: '2a02:ecc0::/29",
+			[]string{
+				"Found a potential improvement: '2a02:ecc0::/29",
+				"Found a potential improvement: '202.196.224.5/32",
+			},
 			counts{
-				1,
-				1,
+				2,
+				2,
 			},
+			"",
 		},
 	}
+
 	// Testing the full content of the difference explanation strings is likely to be
 	// tedious and brittle, so we will just check for some substring.
-	for _, test := range tests {
+	for _, test := range goodTests {
 		t.Run(
 			strings.Join([]string{test.gf, test.db}, " "), func(t *testing.T) {
-				c, diffLines, err := processGeofeed(test.gf, test.db)
+				c, dl, err := processGeofeed(test.gf, test.db)
 				assert.NoError(t, err, "processGeofeed ran without error")
-				assert.Contains(t, diffLines[0], test.dl, "got expected substring")
+				for i, s := range test.dl {
+					assert.Contains(
+						t,
+						dl[i],
+						s,
+						"got expected substring: '%s', substring",
+					)
+				}
 				assert.Equal(t, test.c, c, "processGeofeed returned expected results")
 			},
 		)
 	}
+
+	badTests := []processGeofeedTest{
+		{
+			"test_data/geofeed-missing-fields.csv",
```

Comment:
Correct me if I'm wrong but you forgot to check in this file?

Reply:
Nevermind. I missed it.

Reply:
No, I was right. This file appears to be missing.

Reply:
Oops, that should have generated a failure too ...

Reply:
dec98805bd149bbb857e8d3f52f5474ddda38d7b changes the test to look for a specific error, and commits the missing file